hogvm/typescript/src/stl/date.ts

```diff
@@ -0,0 +1,105 @@
+import { DateTime } from 'luxon'
+export interface HogDate {
+    __hogDate__: true
+    year: number
+    month: number
+    day: number
+}
+export interface HogDateTime {
+    __hogDateTime__: true
+    dt: DateTime
+}
+
+export function isHogDate(obj: any): obj is HogDate {
+    return obj && obj.__hogDate__ && 'year' in obj && 'month' in obj && 'day' in obj
+}
+
+export function isHogDateTime(obj: any): obj is HogDateTime {
+    return obj && obj.__hogDateTime__ && 'dt' in obj
+}
+
+export function toHogDate(year: number, month: number, day: number): HogDate {
+    return {
+        __hogDate__: true,
+        year: year,
+        month: month,
+        day: day,
+    }
+}
+
+export function toHogDateTime(timestamp: number | HogDate, zone?: string): HogDateTime {
+    if (isHogDate(timestamp)) {
+        return {
+            __hogDateTime__: true,
+            dt: DateTime.fromObject(
+                {
+                    year: timestamp.year,
+                    month: timestamp.month,
+                    day: timestamp.day,
+                },
+                { zone: zone || 'UTC' }
+            ),
+        }
+    }
+    return {
+        __hogDateTime__: true,
+        dt: DateTime.fromMillis(timestamp * 1000, { zone: zone || 'UTC' }),
```

Comment:
Ah just noticed this is a datetime object so they are both very custom solutions. This is the first STL work that makes me "concerned" that we will have unexpected behaviours in the future that are hard to debug... That said I don't have a particularly good suggestion outside of "make it more limited" or "only use native objects under the hood (which would likely incur (de)serialization costs...

Reply:
I made both implementations use custom `{__hogDateTime: true, dt: number, zone: str}` object types now. 	I previously used objects in python because python supports classes, whereas for JS everything depends on your bundler (`obj instanceof DateTime` might not work everywhere).